samcli/commands/local/cli_common/options.py

```diff
@@ -121,6 +130,22 @@ def construct_options(f):
             click.option(
                 "--port", "-p", default=port, help="Local port number to listen on (default: '{}')".format(str(port))
             ),
+            click.option(
+                "--ssl-cert-file",
+                default=None,
+                type=click.Path(exists=True),
+                callback=validate_ssl_params,
+                help="Path to SSL certificate file (default: None)",
+            ),
+            # Mark --ssl-key-file as eager, so by the time the --ssl-cert-file validator is invoked,
+            # we know if it's missing
```

Comment:
nit: Another option is to use the [ClickMutex class](https://github.com/aws/aws-sam-cli/blob/adbecfccdee48f5474f4fd3770551d98646cf5ca/samcli/commands/_utils/click_mutex.py#L11) we have to set these two options as required.

An example of this being implemented can be found here: https://github.com/aws/aws-sam-cli/blob/2a527fd18d72e5e7a689e775765cd14e846ea8fe/samcli/commands/init/command.py#L117-L130
if you had wanted to implement it this way.

Reply:
Excellent, I was looking for something like that and didn't see it. I'll look into that in this last round. Thanks!